projects/igniteui-angular/src/lib/directives/mask/mask-parsing.service.ts

```diff
@@ -1,7 +1,18 @@
 import { Injectable } from '@angular/core';
 
-/** @hidden */
-export const MASK_FLAGS = ['C', '&', 'a', 'A', '?', 'L', '9', '0', '#'];
+
+const FLAGS = new Set('aACL09#&?');
+const REGEX = new Map([
+    ['C', /(?!^$)/u], // Non-empty
+    ['&', /[^\p{Separator}]/u], // Whitespace
```

Comment:
I think this shouldn't have the `^` character because as it is, the regex will match anything but a whitespace/invisible separator. 

Also, pressing `space` counts as entering an invisible character and I am not sure if it is intended for the prompt character to be deleted. In my opinion it would make more sense to just move an index forward - as it is right now, though, it will remove the current prompt character.  The same thing goes for all literals that can match a whitespace.

Reply:
That should be `Non-whitespace` but otherwise it follows the [specification](https://github.com/IgniteUI/igniteui-angular/wiki/Mask-Directive-Specification#developer)